src/UI/V2/Dialog.elm

```diff
@@ -0,0 +1,132 @@
+
+
+# Component handling
+
+@docs map
+
+-}
+
+import Element exposing (Element, fill, shrink)
+import Element.Border as Border
+import Element.Events as Events
+import UI.Button as Button exposing (Button)
+import UI.Icon as Icon exposing (Icon)
+import UI.Internal.Colors exposing (mainBackground)
+import UI.Internal.RenderConfig exposing (RenderConfig, localeTerms)
+import UI.Palette as Palette
+import UI.RenderConfig as RenderConfig exposing (RenderConfig)
+import UI.Size as Size
+import UI.Text as Text
+import UI.Utils.ARIA as ARIA exposing (roleButton)
+import UI.Utils.Element as Element
+
+
+
+-- Building
+
+
+{-| The `Dialog msg` type is used for describing the component for later
+rendering.
+-}
+type Dialog msg
+    = Dialog (Properties msg) (Options msg)
+
+
+type alias Properties msg =
+    { title : String
+    , icon : Icon
+    , close : msg
+    }
+
+
+type alias Options msg =
+    { body : Element msg
+    , buttons : List (Button msg)
+    , closeOnOverlayClick : Bool
+    }
+
+
+{-| Constructs a dialog by receiving its title, icon in the title and a close
+message.
+-}
+dialog : String -> Icon -> msg -> Dialog msg
+dialog title icon closeMsg =
+    Dialog (Properties title icon closeMsg) (Options Element.none [] False)
+
+
+
+-- Component handling
+
+
+{-| Transforms the message produced by the component.
+-}
+map : (a -> b) -> Dialog a -> Dialog b
+map applier (Dialog { title, icon, close } { body, buttons, closeOnOverlayClick }) =
+    Dialog
+        { title = title
+        , icon = icon
+        , close = applier close
+        }
+        { body = Element.map applier body
+        , buttons = List.map (Button.map applier) buttons
+        , closeOnOverlayClick = closeOnOverlayClick
+        }
+
+
+
+-- Content
+
+
+{-| With `Dialog.withBody` you can specify the body of the dialog.
+**Note**: By default, the body is `Element.none`
+-}
+withBody : Element msg -> Dialog msg -> Dialog msg
+withBody body (Dialog props options) =
+    Dialog props { options | body = body }
+
+
+{-| With `Dialog.withButtons` you can specify the buttons for the footer of the
+dialog.
+**Note**: By default, the buttons are an empty list
+-}
+withButtons : List (Button msg) -> Dialog msg -> Dialog msg
+withButtons buttons (Dialog props options) =
+    Dialog props { options | buttons = buttons }
+
+
+{-| Allows you to specify whether clicking on overlay closes the dialog or not.
+By default, clicking on overlay does nothing, that is because an unintentional
+click can cause loss of information if a dialog is open which is expecting an
+input from the user and user has already provided it partially. It is
+recommended to provide an explicit cancel/close button for such cases. However
+this helper can be used to recieve clicks on overlay to close the dialog where
+loss of information is not likely to happen.
+-}
+withCloseOnOverlayClick : Bool -> Dialog msg -> Dialog msg
+withCloseOnOverlayClick shouldClose (Dialog props options) =
```

Comment:
I would not publish this one yet.
Adding new exposed functions bumps minor versions, but removing them are considered api-breaking and bumps major versions.